### PR TITLE
[4.0] [com_mails] Prepared statements

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

Adds prepared statements to `com_mails`.

### Testing Instructions

Go to System Dashboard -> Templates -> Mail Configuration.
Use the search and filters.

### Expected result

Works like before.

### Documentation Changes Required

No.

### CODE REVIEW

File: administrator/components/com_mails/Model/TemplatesModel.php [hunk 1]
```diff
@@ -118,54 +120,57 @@ protected function getListQuery()
...
+					->bind([':search1', ':search2', ':search3', ':search4'], $search);
 			}
 		}
 
 		// Filter on the extension.
-		if ($language = $this->getState('filter.extension'))
+		if ($extension = $this->getState('filter.extension'))
 		{
-			$query->where('a.template_id LIKE ' . $db->quote($language . '.%'));
+			$extension = $extension . '.%';
```
Comment (Quy) [inline administrator/components/com_mails/Model/TemplatesModel.php]:
```suggestion
			$extension .= '.%';
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 3db59fd45f0118a62fcfa6fde69b2e85eb2d541c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26438">issues.joomla.org/tracker/joomla-cms/26438</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 3db59fd45f0118a62fcfa6fde69b2e85eb2d541c<br /><br />just one side note why template_id is a varchar ?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26438">issues.joomla.org/tracker/joomla-cms/26438</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26438">issues.joomla.org/tracker/joomla-cms/26438</a>.</sub>

**Comment by SharkyKZ** [comment]
Because it's a string e.g. `com_config.test_mail`.

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**